test/system/075-exec.bats

```diff
@@ -127,4 +127,16 @@ load helpers
     run_podman rm -t 0 -f $cid
 }
 
+@test "podman exec --wait" {
+    skip_if_remote "test is meaningless over remote"
+
+    run_podman run -d --name "wait_container" $IMAGE top
+    cid="$output"
+
+    run_podman exec --wait 5 "wait_container" echo hello
+    is "$output" "hello"
```

Comment:
This doesn't really test anything other than the ability to parse `--wait`. So I started writing what I _thought_ would be a test, except it doesn't seem to work:
```console
$ bin/podman create --name foo quay.io/libpod/testimage:20221018 top
<CID>
$ bin/podman exec --wait 5 foo date
[hangs forever, or at least a minute or two while I report this]
```
I was thinking of using that (a `created` container) to then add expect-failure and expect-message-like "timed out". But as best I can tell, `--wait` is not working.....?

Reply:
@rhatdan @bachradsusi  --wait isn't handling `created-but-not-started` containers atm. For podmansh, should --wait just timeout or attempt to start it?

Reply:
I suggest to take a step back and write down which functionality we're looking for exactly.

`podman wait`, for instance, has an `--ignore` flag which may be helpful but also may not be exactly what's needed.

Reply:
Lets continue to hide this feature and look to add the feature to podman wait.  We can get this PR Merged without the podman wait --timeout XYZ support. I like @edsantiago suggested fix, we should fix podman exec --wait to handle this situation.

Reply:
updated PR to timeout on created-but-not-running containers. PTAL

Reply:
Nice! Then please also update the test:
```bash
@test "podman exec --wait" {
    skip_if_remote "test is meaningless over remote"

    run_podman create --name "wait_container" $IMAGE top
    run_podman 125 exec --wait 5 "wait_container" echo hello
    assert "$output" = "Error: timed out waiting for container to run: wait_container"

    run_podman rm -f wait_container
}
```

Reply:
ack

Reply:
@edsantiago btw, did you also check the ginkgo test?